src/Hilke.KineticConvolution/AlgebraicNumberCalculatorExtensions.cs

```diff
@@ -63,5 +63,99 @@ public static class AlgebraicNumberCalculatorExtensions
 
             return calculator.Sign(number) <= 0;
         }
+
+        public static bool IsSmallerThan<TAlgebraicNumber>(
+            this IAlgebraicNumberCalculator<TAlgebraicNumber> calculator,
+            TAlgebraicNumber number1,
+            TAlgebraicNumber number2)
+        {
+            if (calculator is null)
+            {
+                throw new ArgumentNullException(nameof(calculator));
+            }
```

Comment:
Ce if apparait a pleins d'endroits, en l'encapsulant dans un méthode, on pourrait diminuer le nombre de lignes de code.

Reply:
Normally, we try to avoid using some kind of helper function to test parameters... The only reason is that it distorts the result of the code coverage. For this method, if you don't test the case `if (calculator is null)`, code coverage will tell you that the code inside is not covered. If you use an helper method, code coverage will alway go through this method, and considered this lne as "covered". I'm not sure it's the right way, but it's the right way ;)

Reply:
Effectivement, je prefere ne pas faire de helper juste pour gagner quelques lignes.

Reply:
Well done! I haven't thought about that point.

Reply:
Je resous.